src/main/java/org/candlepin/subscriptions/tally/InventoryAccountUsageCollector.java

```diff
@@ -196,4 +201,27 @@ else if (facts.isVirtual() && !StringUtils.isEmpty(facts.getHypervisorUuid())) {
         return calcsByAccount;
     }
 
+    private List<Host> getAccountHosts(Collection<String> accounts) {
+        return accounts.stream()
+            .map(hostRepository::findByAccountNumber)
+            .flatMap(List::stream)
+            .collect(Collectors.toList());
+    }
+
+    private Host getOrCreateHost(Map<String, Host> inventoryHostMap, InventoryHostFacts hostFacts,
+        NormalizedFacts facts) {
+
+        Host existingHost = inventoryHostMap.remove(hostFacts.getInventoryId().toString());
```

Comment:
:worried: Not sure how I feel about the side-effects of this method.  Is there a way we could break this up into two steps?  Something like 

```
getOrCreateHost(update(inventoryHostMap, hostFacts), hostFacts, facts);
```

Where `update` just does the removal?  It's not as pretty to look at, but it makes it more obvious that the map is being mutated.

Reply:
@awood @lindseyburnett I've refactored a bit. Please take a look when you get a chance.